src/modules/url-changer.js

```diff
@@ -315,75 +170,106 @@
+					currentPageFragment = currentPageFragment.substring(0, currentQsIndex + 1) +
 						newQsString;
 				}
 				
 				//_currentPage
-				_currentStrategy.updateUrlFragment();
-				$.sendPageView({page: data.route});
+				updateUrlFragment();
+
+				if (options.raiseFragmentChanged) {
+					App.mediator.notify('page.fragmentChanged', currentPageFragment);
+				}
+
+				//??? options.route doesnt exists....
+				$.sendPageView({page: options.route});
 			}
 		}
 	};
 	
+	var urlChanged = function () {
+		var nextPage = App.pages.page(window.location.pathname);
+		
+		//if we found a page for this route
+		if (nextPage) {
+			var loc = window.location;
+			//Detect if we change page
+			if (nextPage.key() == currentPageKey) {
+				if (!loc.origin) {
+					// IE !!
+					loc.origin = loc.protocol + '//' + loc.hostname;
+				}
+				var cur = loc.origin + currentPageRoute;
+				var pageFragment = loc.href.substring(cur.length);
+				
+				if (currentPageFragment != pageFragment) {
+					App.mediator.notify('page.fragmentChanged', pageFragment);
+					currentPageFragment = pageFragment;
+					//Sync Qs
+					extractQS();
+				} else {
+					App.mediator.notify('page.sameFragmentRequested', pageFragment);
+				}
+			} else {
+				isPopingState = true;
+				App.mediator.goto(loc.pathname + loc.search, currentPageUrl);
+			}
+		} else {
+			App.log({args: 'Page not found', me: 'Url Changer'});
+		}
+	};
+
 	var onNavigateToCurrent = function (key, data, e) {
-		var _oldFragment = _currentPageFragment;
+		var oldFragment = currentPageFragment;
 		
-		_extractFragmentFromRoute(_getNextRouteFromData(data), data.route);
-		_currentStrategy.updateUrlFragment();
+		extractFragmentFromRoute(getNextRouteFromData(data), data.route);
+		updateUrlFragment();
 		
 		//Set back old fragment to trigger fragment changed
-		_currentPageFragment = _oldFragment;
-		_extractQS();
-		_currentStrategy.urlChanged();
+		currentPageFragment = oldFragment;
+		urlChanged();
 		$.sendPageView({page: data.route});
 	};
 	
-	var _urlChanged = function () {
-		_currentStrategy.urlChanged();
-	};
-	
 	var getCurrentUrl = function () {
-		return _currentStrategy.getCurrentUrl();
+		return window.location.pathname;
 	};
 	
 	var getQueryString = function () {
-		return _currentStrategy.getQueryString();
+		return window.location.search;
 	};
+
 	var getFullUrl = function () {
-		return _currentStrategy.getFullUrl();
+		return window.location.toString();
 	};
 	
 	var init = function () {
 		//Detect good strategy
 		if (window.history.pushState) {
-			_currentStrategy = _strategies.history;
-			win.on('popstate', _urlChanged);
-			
-		} else if ($.bbq) {
-			
-			_currentStrategy = _strategies.hash;
-			//Attach to hash change
-			win.on('hashchange', _urlChanged);
+			win.on('popstate', urlChanged);
 		} else {
 			App.log({
 				fx: 'error',
-				msg: 'Cannot update url : history api and bbq are not found'
+				msg: 'Cannot update url : history api was not found'
```

Comment:
@PascalPiche Please tell me over slack which projects runs this version. I'll debug it myself. Thanks.